Framework/PythonInterface/mantid/utils/absorptioncorrutils.py

```diff
@@ -32,174 +33,215 @@ def _getBasename(filename):
+                "sample_formula": ws.run()['SampleFormula'].lastValue().strip(),
+                "mass_density": ws.run()['SampleDensity'].lastValue(),
+                "height_unit": ws.run()['BL11A:CS:ITEMS:HeightInContainerUnits'].lastValue(),
+                "height": ws.run()['BL11A:CS:ITEMS:HeightInContainer'].lastValue(),
+                "sample_container": ws.run()['SampleContainer'].lastValue().replace(" ", ""),
+                "abs_method": abs_method,
+            }.items()
+        ]
+
+        # use mantid build-in alg to generate the cache filename and sha1
+        cache_filename, signature = mantid.simpleapi.CreateCacheFilename(
+            Prefix=prefix_name,
+            OtherProperties=property_string,
+            CacheDir=cache_dir,
+        )
+
+    return cache_filename, signature
+
+
+def __load_cached_data(cache_file_name, sha1, abs_method="", prefix_name=""):
+    """try to load cached data from memory and disk
+
+    :param abs_method: absorption calculation method
+    :param sha1: SHA1 that identify cached workspace
+    :param cache_file_name: cache file name to search
+    :param prefix_name: prefix to add to wkspname for caching
+
+    return  found_abs_wksp_sample, found_abs_wksp_container
+            abs_wksp_sample, abs_wksp_container
     """
-    wsn_as = ""  # absorption workspace sample
-    wsn_ac = ""  # absorption workspace container
-    found_as = False
-    found_ac = False
+    # init
+    abs_wksp_sample, abs_wksp_container = "", ""
+    found_abs_wksp_sample, found_abs_wksp_container = False, False
 
     # step_0: depending on the abs_method, suffix will be different
     if abs_method == "SampleOnly":
-        wsn_as = f"{absName}_ass"
-        found_ac = True
+        abs_wksp_sample = f"{prefix_name}_ass"
+        found_abs_wksp_container = True
     elif abs_method == "SampleAndContainer":
-        wsn_as = f"{absName}_ass"
-        wsn_ac = f"{absName}_acc"
+        abs_wksp_sample = f"{prefix_name}_ass"
+        abs_wksp_container = f"{prefix_name}_acc"
     elif abs_method == "FullPaalmanPings":
-        wsn_as = f"{absName}_assc"
-        wsn_ac = f"{absName}_ac"
+        abs_wksp_sample = f"{prefix_name}_assc"
+        abs_wksp_container = f"{prefix_name}_ac"
     else:
         raise ValueError("Unrecognized absorption correction method '{}'".format(abs_method))
 
-    # step_1: check memory to see if the ws is already there
-    # -- check SHA1
-    # NOTE: The new attribute absSHA1 should be there.
-    if mtd.doesExist(wsn_as):
-        found_as = mtd[wsn_as].run()["absSHA1"].value == sha1
-    if mtd.doesExist(wsn_ac):
-        found_ac = mtd[wsn_ac].run()["absSHA1"].value == sha1
-
-    # step_2: try to load from cache_file provided
-    if (not found_as) or (not found_ac):
-        # disk -> memory
+    # step_1: check memory
+    if mtd.doesExist(abs_wksp_sample):
+        found_abs_wksp_sample = mtd[abs_wksp_sample].run()["absSHA1"].value == sha1
```

Comment:
If caching is enabled, the absorption workspace generated here will always have ["absSHA1"].

If the user really want to break the code, they can do the following
- run `SNSPowderReduction` with caching enabled so that they can record the 40 char sha
- make a mock workspace in memory that has the same name (instrument_sha_abs_correction_ass)
- run `SNSPowderReduction` again so that the mocked workspace can trigger the KeyNotFoundError

Given the peculiarity of this kind of edge case, I don't think we need to check if the log entry exists or not.